src/reason_parser.mly

```diff
@@ -3196,6 +3196,13 @@ mark_position_pat
   | pattern_without_or AS as_loc(error)
     { expecting_pat (with_txt $3 "identifier") }
 
+  /* | Foo () => ... is sugar for | Foo (()) => ...*/
+  /* putting it here instead of in pattern_constructor_argument because that one might cause too many shift/reduce conflicts... */
+  | as_loc(constr_longident) LPAREN RPAREN
+    {
+      let loc = mklocation $startpos $endpos in
```

Comment:
@jordwalke I have no idea what I'm doing with location here. Is this right?

Reply:
I believe this is okay.